### PR TITLE
List structure

### PR DESCRIPTION
**Pull request by mmsesay**
In the PR the following was done:

- Setup webpack and css loader
- Setup linters configuration
- Add an array of objects with tasks
- Created a function to iterate through the tasks and render in the HTML element
- Invoke the function on page load

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
```
Comment (rloterh) [inline webpack.config.js:4]: - [x] _Mode_ option is not set for _webpack_. Kindly set the `mode` and either assign its value to `development` or `production` within the _module.exports_

**Webpack config error:**
image

File: package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "todo-list-project",
+  "version": "1.0.0",
+  "description": "this is a simple todo list application using webpack and it is being served by webpack dev server",
+  "private": true,
+  "scripts": {
```
Comment (rloterh) [inline package.json:6]: - [ ] A local server cannot be started within your application. Please add start options to your scripts to enable your application to be run using the command `npm start`, `npm run start` or `npm serve`.

**Package config error:**
image

File: src/styles/style.css [hunk 3]
```diff
@@ -0,0 +1,58 @@
+body {
```
Comment (rloterh) [inline src/styles/style.css:1]: - [ ] OPTIONAL
Your application is not properly laid out in desktop versions. It appears at the bottom of the page. Kindly improve upon this aspect by ensuring that your app is accessible without scrolling the page.

**Your Application desktop view:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Muhammad, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: webpack.config.js (hunk 1 above)
Comment (mmsesay) [reply webpack.config.js:4]: Thanks for picking this out. It has been resolved

File: package.json (hunk 2 above)
Comment (mmsesay) [reply package.json:6]: Thanks for picking this out. It has been resolved

File: src/styles/style.css (hunk 3 above)
Comment (mmsesay) [reply src/styles/style.css:1]: Thanks for picking this out. It has been resolved

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  entry: './src/js/index.js',
+  mode: 'development',
+  plugins: [
```
Comment (rloterh) [inline webpack.config.js:7]: - [ ] OPTIONAL
Kindly specify a `dev Sever` in your `Webpack.config.js` like this:

**Add a devServer to your Webpack:**
image

File: package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack-dev-server --mode production --entry ./src/js/index.js --output-filename ./dist/bundle.js",
+    "build:prod": "webpack src/js/index.js dist/bundle.js -p",
+    "start": "webpack serve"
```
Comment (rloterh) [inline package.json:10]: - [x] OPTIONAL
Your local server does open automatically when the command `npm start` is run. Kindly modify the start option to be:
`"start": "webpack serve --open",` 
to enable the page load automatically in the browser.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Muhammad, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾:shipit:.
Congratulations! 🎉🥂

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comment(s) marked **[OPTIONAL]** under this review:_

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 5 above)
Comment (mmsesay) [reply package.json:10]: I have added the --open to my start script

File: webpack.config.js (hunk 4 above)
Comment (mmsesay) [reply webpack.config.js:7]: Thanks. The dev server has been included.

**Pull request merged**